bundles/org.openhab.core.crypto.cryptostore/src/main/java/org/openhab/core/crypto/cryptostore/CryptoStore.java

```diff
@@ -0,0 +1,328 @@
+    private int keyLength = 2048;
+    private String alias = "openhab";
+    private String caAlias = "trustedCa";
+    private String distName = "CN=openHAB, O=openHAB, L=None, ST=None, C=None";
+    private String cipher = "AES/GCM/NoPadding";
+    private String keyAlgorithm = "";
+
+    private @Nullable Cipher encryptionCipher;
+
+    public CryptoStore() {
+        try {
+            encryptionCipher = Cipher.getInstance(cipher);
+        } catch (NoSuchAlgorithmException | NoSuchPaddingException e) {
+            logger.debug("Could not get cipher instance", e);
+        }
+    }
+
+    public byte[] generateEncryptionKey() {
+        Key key;
+        try {
+            KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+            keyGenerator.init(keySize);
+            key = keyGenerator.generateKey();
+            byte[] newKey = key.getEncoded();
+            this.keyAlgorithm = key.getAlgorithm();
+            return newKey;
+        } catch (NoSuchAlgorithmException e) {
+            logger.debug("Could not generate encryption keys", e);
+        }
+        return new byte[0];
+    }
+
+    private Key convertByteToKey(byte[] keyString) {
+        Key key = new SecretKeySpec(keyString, keyAlgorithm);
+        return key;
+    }
+
+    public String encrypt(String data, Key key) throws Exception {
+        return encrypt(data, key, this.cipher);
+    }
+
+    public String encrypt(String data, Key key, String cipher) throws Exception {
+        byte[] dataInBytes = data.getBytes();
+        Cipher encryptionCipher = this.encryptionCipher;
+        if (encryptionCipher != null) {
+            encryptionCipher.init(Cipher.ENCRYPT_MODE, key);
+            byte[] encryptedBytes = encryptionCipher.doFinal(dataInBytes);
+            return Base64.getEncoder().encodeToString(encryptedBytes);
+        } else {
+            return "";
+        }
+    }
+
+    public String decrypt(String encryptedData, Key key) throws Exception {
+        return decrypt(encryptedData, key, this.cipher);
+    }
+
+    public String decrypt(String encryptedData, Key key, String cipher) throws Exception {
+        byte[] dataInBytes = Base64.getDecoder().decode(encryptedData);
+        Cipher decryptionCipher = Cipher.getInstance(cipher);
+        Cipher encryptionCipher = this.encryptionCipher;
+        if (encryptionCipher != null) {
+            GCMParameterSpec spec = new GCMParameterSpec(keySize, encryptionCipher.getIV());
+            decryptionCipher.init(Cipher.DECRYPT_MODE, key, spec);
+            byte[] decryptedBytes = decryptionCipher.doFinal(dataInBytes);
+            return new String(decryptedBytes);
+        } else {
+            return "";
+        }
+    }
+
+    public void setPrivKey(String privKey, byte[] keyString) throws Exception {
+        Key key = convertByteToKey(keyString);
+        this.privKey = encrypt(privKey, key);
+    }
+
+    public String getPrivKey(byte[] keyString) throws Exception {
```

Comment:
Wouldn't it be better to have methods for sign/verify and encrypt/decrypt in this class instead of exposing the private key? Or what other use case would a binding developer have to use the key?

Reply:
I added those into this in the event someone had a use case where they needed to access their private keys.  You're absolutely right, it should be held in the class IMHO.  I don't use these functions in ATV.  I was just trying to be complete by providing access should someone have a use case.  I can pull them out if they aren't needed.